DemoProject/DemoProject/MapViewController2.swift

```diff
@@ -0,0 +1,328 @@
+    
+    private func setUpUI() {
+        mapView.showsUserLocation = true
+        mapView.delegate = self
+        let overlay = CustomOverlay(coordinates: self.coordinates, count: coordinates.count)
+        mapView.addOverlay(overlay)
+        mapView.preferredConfiguration = MKHybridMapConfiguration()
+        mapView.isScrollEnabled = true
+        mapView.isZoomEnabled = true
+        mapView.isPitchEnabled = true
+        mapView.showsBuildings = true
+        myRegion.delegate = self
+        myRegion.requestAlwaysAuthorization()
+        /** Creating center camera view */
+//        let mapcamera = MKMapCamera(lookingAtCenter: CLLocationCoordinate2D(latitude: hotelLocation.coordinate.latitude, longitude: hotelLocation.coordinate.longitude), fromDistance: 500, pitch: 60, heading: 0.0)
+//                mapView.camera = mapcamera
+//        mapView.setCamera(mapcamera, animated: true)
+        mapView.register(MyAnnotationView.self, forAnnotationViewWithReuseIdentifier: MKMapViewDefaultAnnotationViewReuseIdentifier)
+        
+        checkLocationServices()
+        centerLocation()
+        cameraBoundary()
+        //this is normal annotation with title and subtitle
+//        addAnnotation()
+        addCustomAnnotation()
+        showRouteOnMap(pickupCoordinate: hotelAnnotation.coordinate, destinationCoordinate: bakeryAnnotation.coordinate)
+        createRegion()
+        let distance = bakeyShopLocation.distance(from: hotelLocation) / 1000
+        print("Distance = \(String(format:"%.02f", distance)) KMs")
+    }
+    
+    func createRegion() {
+        let myCoordinate = CLLocationCoordinate2D(latitude: 22.274992457213102, longitude: 70.80532688859402)
+        let region = CLCircularRegion(center: myCoordinate, radius: 100, identifier: "myArea")
+        region.notifyOnEntry = true
+        region.notifyOnExit = true
+        myRegion.startMonitoring(for: region)
+    }
+    
+    func checkLocationServices() {
+        if CLLocationManager.locationServicesEnabled() {
+            self.checkLocationAuth()
+        }
+    }
+    
+    func checkLocationAuth() {
+        switch CLLocationManager.authorizationStatus() {
+        case .authorizedAlways:
+            break
+        case .authorizedWhenInUse:
+            mapView.showsUserLocation = true
+            break
+        case . denied:
+            break
+        case .notDetermined:
+            locationManager.requestAlwaysAuthorization()
+            mapView.showsUserLocation = true
+            break
+        default:
+            break
+        }
+
+        locationManager.delegate = self
+        locationManager.startUpdatingHeading()
+    }
+    
+    func centerLocation() {
+        let region = MKCoordinateRegion(center: initialLocation.coordinate, latitudinalMeters: 500, longitudinalMeters: 500)
+        mapView.setRegion(region, animated: true)
+    }
+    
+    func cameraBoundary() {
+        /** Setting initial view how much portion to display on map **/
```

Comment:
instead of commenting whole code in function just comment the method where u are calling and add the description there